src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1077,6 +1103,15 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 			}
 
 			queryTableCount++;
+
+			/* we do not expect to see a view in modify query */
+			if (rangeTableEntry->relkind == RELKIND_VIEW)
+			{
+				ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+								errmsg("cannot perform distributed planning for the "
```

Comment:
the errmsg is almost useless - the detail already seems like a better message. Something like "cannot modify views including distributed tables" would be better